File: chess_cpp/validators/PathAvailableValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "PathAvailableValidator.h"
#include "../utils/MessageHandler.h"

bool PathAvailableValidator::validate(Board *board, Spot *start, Spot *end) {
    std::vector<Spot> onPathSpots = start->getPiece().get()->getPathSpots(board, start, end);
    return onPathSpots.size() == 0;
}

std::string PathAvailableValidator::getNotValidMessage() {
    return MessageHandler::getNoMessage();
}

File: chess_java/src/chess/validators/PathAvailableValidator.java
package chess.validators;


import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

import java.util.List;

public class PathAvailableValidator extends PieceMoveValidator {


    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        List<Spot> onPathSpots = start.getPiece().getPathSpots(board, start, end);
        return onPathSpots == null || onPathSpots.isEmpty();
    }

    @Override
    public String getNotValidMessage() {
        return MessageHandler.getNoMessage();
    }
}
